### ISSUE TITLE
How to define variable reused by different tasks?

### ISSUE THREAD

**Issue by northtree** [illBeRoy/taskipy#37]
For instance, I would like to run `lint` and `black` on same multiple folders, instead of specify multiple times.

```
[tool.taskipy.tasks]
lint = "pylint -r n foo bar"
black = "black foo bar"
```

=> 
 
```
[tool.taskipy.tasks]
path = "foo bar"
lint = "pylint -r n path"
black = "black path"
```

**Comment by Granitosaurus** [comment]
There are few ways to do this
1. For your example you can just use args feature:

```
[tool.taskipy.tasks]
lint = "pylint -r n"
black = "black"
```

Then run it: `poetry run task black SOMEPATH` or `poetry run task lint SOMEPATH`.  SOMEPATH will essential be appended to the command: `black SOMEPATH`.

2. Use environment variables:

```
[tool.taskipy.tasks]
lint = "pylint -r n $PATH"
black = "black $PATH"
```

Then run it: `PATH=mypath poetry run task black` (for bash) or set PATH before running command (via .env or something)

**Comment by illBeRoy** [comment]
Following @Granitosaurus' great answer, you can load environment variables automatically using dotenv. You can learn more about this in our [advanced features doc](https://github.com/illBeRoy/taskipy/blob/master/docs/ADVANCED_FEATURES.md).

Does this help? 🙂

**Comment by northtree** [comment]
Thanks for sharing. It would be better if supported within this setting section.

> [tool.taskipy.settings]

**Comment by illBeRoy** [comment]
## TL;DR
I think that if we introduce variables into the system, it should probably make use of `dotenv` behind the scenes. This means that if you want to declare variables, you will have to include a `.env` file in your project similar to:

```env
PATH=foo bar
OTHER_VAR=a
```

and so on and so forth.

## Long story (or, what made me come to this conclusion)
Suppose that we add a variables system, and we put it under the `pyproject.toml` file. How would we pass the variables into the scripts? We could:
1. Render them into the string
2. Pass them as environment variables

Option 1 contradicts the main selling point of taskipy: simplicity. Every task is a simple, straight forward shell command. Once we introduce templating, the tasks are no longer that, which can lead to unexpected behavior (e.g. dev mistakenly includes a non-escaped character in their task).

If we go with Option 2, though, we essentially get a simple version of `dotenv`. This begs the question - why not just use dotenv? It's standard, it's used over a multitude of stacks and languages, it's supported on both dev and production, and whoever used dotenv in the past will feel right at home :)

What do you guys think?

**Comment by Granitosaurus** [comment]
I kind agree that dotenv integrates nicely!   
However I'm not sure I'd agree it's the simpler path - it's simple for people who _already_ use/know dotenv. I guess if you were to take an objective stance pyproject is more _pythonic_ than dotenv, however dotenv seems to be much more practical here as tasks often rely on some environment variables anyway.

Maybe we should move the Advanced Usage doc to README.md as it seems like important piece of knowledge and leave it at that.

**Comment by northtree** [comment]
I vote for @Granitosaurus 's PR #38 for easier to integrate within CI (also not dependent on `dotenv`).

**Comment by illBeRoy** [comment]
After taking my time to think about it, we can actually go with @Granitosaurus proposal, but will need to expend upon it:
1. The variables should be declared in a dedicated section in the toml file
2. The feature should be **opt in**, and not apply automatically and unbeknownst to the user :)

Therefore, how about the following design:

Variables will be declared in their own section (not settings):
```toml
[tool.taskipy.variables]
path = 'foo bar'
```

And we will support templating only if the user explicitly marked the task as such, using the [explicit task syntax](https://github.com/illBeRoy/taskipy#adding-tasks):
```toml
[tool.taskipy.tasks]
lint = { cmd = "pylint -r n {path}", use_vars = True }
black = { cmd = "black {path}", use_vars = True }
```

**Comment by Granitosaurus** [comment]
@illBeRoy great idea! Let me adjust the PR :)

**Comment by illBeRoy** [comment]
@northtree 

The feature was merged and released together with [taskipy 1.9.0](https://pypi.org/project/taskipy/1.9.0/).
Please try it out at your leisure. Your feedback would be much appreciated!

**Comment by northtree** [comment]
@illBeRoy Just checkout 1.9.0, LGTM.

BTW, please note that you have typos on previous comment, which should be `use_vars = true`.

Otherwise will raise `Invalid TOML file` error. 
> use_vars = True

```
      30│     def read(self):  # type: () -> "TOMLDocument"
      31│         try:
      32│             return super(TOMLFile, self).read()
      33│         except (ValueError, TOMLKitError) as e:
    → 34│             raise TOMLError("Invalid TOML file {}: {}".format(self.path.as_posix(), e))
      35│ 
      36│     def __getattr__(self, item):  # type: (str) -> Any
      37│         return getattr(self.__path, item)
      38│ 

```

At last, just noticed that you did provide excellent README though. Thanks for your awesome work. @Granitosaurus @illBeRoy 🚀
